code/modules/mob/inventory.dm

```diff
@@ -185,9 +185,22 @@ var/list/slot_equipment_priority = list( \
 	return drop_from_inventory(r_hand, target)
 
 //Drops the item in our active hand. TODO: rename this to drop_active_hand or something
-/mob/proc/drop_item(var/atom/target)
-	if(hand)	return drop_l_hand(target)
-	else		return drop_r_hand(target)
+/mob/proc/drop_item(var/atom/Target)
+	if(hand)
+		make_item_drop_sound()
+		return drop_l_hand(Target)
+	else
+		make_item_drop_sound()
+		return drop_r_hand(Target)
+
+/mob/proc/make_item_drop_sound()
+	var/obj/item/I = get_active_hand()
+	spawn (1)
```

Comment:
oh no. Bad memes detected.

First, `I` is known to us from the calling code. Simply pass it in as an argument.
Second, use a timer instead of spawn.
Third, rework drop_item to be less silly.

```dm
//Drops the item in our active hand. TODO: rename this to drop_active_hand or something
/mob/proc/drop_item(var/atom/target)
	if (istype(target, /obj/item))
		addtimer(CALLBACK(src, .proc/make_item_drop_sound, target), 1)

	if (hand)
		return drop_l_hand(target)
	else
		return drop_r_hand(target)

/mob/proc/make_item_drop_sound(obj/item/target)
	if (!istype(target) && !QDELETED(target))
		return

	if (target.drop_sound)
		playsound(target, target.drop_sound, 25, 0)

	target.visible_message("<small>\the [target] falls to the floor.</small>")
```